await-generator/src/SOFe/AwaitGenerator/PubSub.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * await-generator
+ *
+ * Copyright (C) 2023 SOFe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace SOFe\AwaitGenerator;
+
+use RuntimeException;
+
+/**
+ * A pubsub allows coroutines to publish a message that is received by all subscribres.
+ * @template T
+ */
+final class PubSub{
+	/** @var Channel<T>[] */
+	private array $subscribers = [];
+
+	/**
+	 * If a subscriber is lagging for more than $maxLag items behind,
+	 * a RuntimeException is thrown to help detect memory leaks in advance.
+	 * Set it to `null` to disable the check.
+	 */
+	public function __construct(
+		private ?int $maxLag = 10000,
+	) {}
+
+	/**
+	 * Publishes a message and return.
+	 *
+	 * This method does not block until the event is actually received by subscribers.
```

Comment:
```suggestion
	 * This method does not wait for the event to be actually received by subscribers.
```
Took a while to realize.